frontend/lib/src/enums/transaction_enum.dart

```diff
@@ -3,76 +3,26 @@
 enum TransactionTypes { CREDIT, DEPT, TRANSFER }
 
 enum TransactionCategories {
-  FOOD,
-  TRANSPORTATION,
-  BILLS,
-  SAVINGS,
-  SALARY,
-  SENDER,
-  RECIPIENT,
-  MISC,
-}
+  FOOD('Food/Drinks'),
+  TRANSPORTATION('Transportation'),
+  BILLS('Housing/Billings'),
+  SAVINGS('Savings'),
+  SALARY('Salary'),
+  SENDER('Sender'),
+  RECIPIENT('Recipient'),
+  MISC('Miscellaneous');
 
-extension TransactionCategoriesExtension on TransactionCategories {
-  String get value {
-    switch (this) {
-      case TransactionCategories.FOOD:
-        return 'Food/Drinks';
-      case TransactionCategories.TRANSPORTATION:
-        return 'Transportation';
-      case TransactionCategories.BILLS:
-        return 'Housing/Billings';
-      case TransactionCategories.SAVINGS:
-        return 'Savings';
-      case TransactionCategories.MISC:
-        return 'Miscellaneous';
-      default:
-        return '';
-    }
-  }
-}
+  final String value;
+  const TransactionCategories(this.value);
 
-List<String> creditCategories = TransactionCategories.values
-    .where(
-  (value) =>
-      value == TransactionCategories.FOOD ||
-      value == TransactionCategories.TRANSPORTATION ||
-      value == TransactionCategories.BILLS ||
-      value == TransactionCategories.SAVINGS ||
-      value == TransactionCategories.MISC,
-)
-    .map((value) {
-  switch (value) {
-    case TransactionCategories.FOOD:
-      return 'Food/Drinks';
-    case TransactionCategories.TRANSPORTATION:
-      return 'Transportation';
-    case TransactionCategories.BILLS:
-      return 'Housing/Billings';
-    case TransactionCategories.SAVINGS:
-      return 'Savings';
-    case TransactionCategories.MISC:
-      return 'Miscellaneous';
-    default:
-      return '';
-  }
-}).toList();
+  static List<String> creditCategories = [
+    TransactionCategories.FOOD.value,
+    TransactionCategories.TRANSPORTATION.value,
+    TransactionCategories.BILLS.value,
+    TransactionCategories.SAVINGS.value,
+    TransactionCategories.MISC.value,
+  ].toList();
 
-TransactionCategories transactionCategoriesFromString(
-  String creditTransaction,
-) {
-  switch (creditTransaction) {
-    case 'Food/Drinks':
-      return TransactionCategories.FOOD;
-    case 'Transportation':
-      return TransactionCategories.TRANSPORTATION;
-    case 'Housing/Billings':
-      return TransactionCategories.BILLS;
-    case 'Savings':
-      return TransactionCategories.SAVINGS;
-    case 'Miscellaneous':
-      return TransactionCategories.MISC;
-    default:
-      return TransactionCategories.MISC;
-  }
+  static TransactionCategories fromValue(String jsonValue) =>
+      TransactionCategories.values.singleWhere((i) => jsonValue == i.value);
```

Comment:
```suggestion
  static TransactionCategories fromJson(String jsonValue) =>
      TransactionCategories.values.singleWhere((i) => jsonValue == i.name);
      
       static TransactionCategories fromValue(String value) =>
      TransactionCategories.values.singleWhere((i) => value == i.value);
```

then add some comments on the static constructor